src/scripts/main.js

```diff
@@ -1,3 +1,1004 @@
+        workingArray[currentCellIndex] = null;
+        changeIndicator = true;
+
+        if (workingArray[indexThroughTwoCells] === null) {
+          page.querySelectorAll('.field-cell')[indexThroughTwoCells]
+            .classList.add(currentModifier);
+
+          page.querySelectorAll('.field-cell')[indexThroughOneCell]
+            .classList.remove(currentModifier);
+
+          workingArray[indexThroughTwoCells]
+          = workingArray[indexThroughOneCell];
+          workingArray[indexThroughOneCell] = null;
+          changeIndicator = true;
+        }
+      } else if (workingArray[indexThroughTwoCells] === null) {
+        page.querySelectorAll('.field-cell')[indexThroughTwoCells]
+          .classList.add(currentModifier);
+
+        page.querySelectorAll('.field-cell')[currentCellIndex]
+          .classList.remove(currentModifier);
+        workingArray[indexThroughTwoCells] = workingArray[nearestCellIndex];
+        workingArray[currentCellIndex] = null;
+        changeIndicator = true;
+      }
+    };
+  };
+
+  winningMessage();
+
+  if (changeIndicator) {
+    generateTwoOrFour();
+  }
+
+  loseMessage();
+};
+
+function downwardMovement() {
+  if (disableFunctions) {
+    return;
+  }
+
+  changeIndicator = false;
+
+  let allIndices = [];
+
+  allIndices = [];
+
+  for (let i = workingArray.length - 1; i >= 0; i--) {
+    if (workingArray[i] !== null) {
+      allIndices.push(i);
+    }
+  }
+
+  for (let i = 0; i < allIndices.length; i++) {
+    const currentCellIndex = allIndices[i];
+    const nearestCellIndex = allIndices[i] + 4;
+    const indexThroughOneCell = allIndices[i] + 8;
+    const indexThroughTwoCells = allIndices[i] + 12;
+
+    const sideBySideAndEqual = ((workingArray[nearestCellIndex] !== null)
+    && (workingArray[nearestCellIndex] === workingArray[currentCellIndex]));
+    const throughOneAndEqual = ((workingArray[indexThroughOneCell] !== null)
+    && (workingArray[indexThroughOneCell] === workingArray[currentCellIndex]));
+    const throughTwoAndEqual = ((workingArray[indexThroughTwoCells] !== null)
+    && (workingArray[indexThroughTwoCells] === workingArray[currentCellIndex]));
+
+    const currentModifier = `field-cell--${workingArray[currentCellIndex]}`;
+    const changeableModifier
+    = `field-cell--${workingArray[currentCellIndex] * 2}`;
+
+    if (sideBySideAndEqual) {
+      page.querySelectorAll('.field-cell')[nearestCellIndex]
+        .classList.remove(currentModifier);
+
+      page.querySelectorAll('.field-cell')[nearestCellIndex]
+        .classList.add(changeableModifier);
+
+      page.querySelectorAll('.field-cell')[currentCellIndex]
+        .classList.remove(currentModifier);
+      workingArray[nearestCellIndex] *= 2;
+      workingArray[currentCellIndex] = null;
+      changeIndicator = true;
+      scoreCounter += workingArray[nearestCellIndex];
+      switcherScore.textContent = scoreCounter;
+
+      if (workingArray[indexThroughOneCell] === null) {
```

Comment:
Same here and for all such cases with `null` as it's a falsy value
```suggestion
      if (!workingArray[indexThroughOneCell]) {
```

Reply:
Hello! I tried to substitute the value (!workingArray[indexThroughOneCell]) instead of (workingArray[indexThroughOneCell] === null). By analogy, I did it everywhere. Unfortunately, the program has completely stopped working.

The matter is that check on 'null' is necessary to me. I am filling all empty cells of the working field with null values. All values outside of the working array are undefined. For the program to work correctly, I'm looking for exactly the null value. If (workingArray[indexThroughOneCell] === null) then we move the cell and sum the value. I deliberately fill empty array cells with 'null' values to distinguish them from external values - undefined. Therefore, I'm looking for exactly null, and if the value is null, then we move the cell inside the square. External values will not be affected because they have a value - undefined.

![Снимок](https://github.com/mate-academy/js_2048_game/assets/121498080/15e35c91-1d06-46a6-bd9f-bbefef8cedd3)